src/braket/experimental/autoqasm/operators/utils.py

```diff
@@ -0,0 +1,45 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+"Utility methods for operators."
+
+from typing import Any
+
+from braket.circuits import FreeParameter
+from braket.experimental.autoqasm import program
+from braket.experimental.autoqasm import types as aq_types
+
+
+def _register_and_convert_parameters(*args: list[Any]) -> list[aq_types.FloatVar]:
```

Comment:
This has side effects. Usually I wouldn't be too much of a fan of compound side effect + return val functions like this, but it encapsulates free parameter handling for all the operator code. Also keeps the other code cleaner.